uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeavePeriodEntitlement.php

```diff
@@ -473,4 +524,51 @@ private function getContractDatesForContactInPeriod(AbsencePeriod $absencePeriod
 
     return $contractsDates;
   }
+
+	/**
+	 * Returns formatted result for getting the balance for an entitlement period given an
+	 * Entitlement Id or (Contact ID + Absence Period ID), When params contains the include_future parameter and its true,
+	 * It returns also future balance for an entitlement taking the Awaiting Approval and More Information Requested leave statuses
+	 * into consideration
+	 *
+	 * Sample param: $params = ['entitlement_id'=>1, 'contact_id'=>9, 'include_future'=>false]
+	 *
+	 * @param $params
+	 * @return an array of formatted results
+	 *
+	 * [
+	 * ['id'=>1, 'remainder'=>['current'=>30, 'future'=>20]]
+	 * ]
+	 */
+	public static function getLeavePeriodEntitlementRemainder($params){
+
+	  $results = [];
+    if($params['entitlement_id']){
```

Comment:
This is dangerous. If the array doesn't have a key named 'entitlement_id' you will get an E_NOTICE. You have to always check if a key exists before trying to access it.